src/analysisd/format/json_extended.c

```diff
@@ -190,15 +221,20 @@ int W_isRootcheck(cJSON *root){
 
  }
  
-// STRTOK every "-" delimiter to get differents groups to our json array.
- void W_JSON_ParseGroups(cJSON *root, const Eventinfo *lf){
+// STRTOK every "-" delimiter to get differents groups to our json array. 
+ void W_JSON_ParseGroups(cJSON *root, const Eventinfo *lf, int nested){
 	cJSON *groups;
 	cJSON *rule;
-	rule = cJSON_GetObjectItem(root,"rule");
+	 
+	if(!nested)
+		rule = root;
+	else
+		rule = cJSON_GetObjectItem(root,"rule");
+
 	cJSON_AddItemToObject(rule,"groups", groups = cJSON_CreateArray());
-	
-	char buffer[strlen(lf->generated_rule->group)];
-	strcpy(buffer, lf->generated_rule->group);	
+	 
+	char buffer[MAX_STRING];
+	strncpy(buffer, lf->generated_rule->group, sizeof(buffer));	

```

Comment:
Should this be `sizeof(buffer) - 1` to ensure it's NUL terminated?

Reply:
Good catch!